src/helpers/HookConfigLib.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+import {HookConfig, ModuleEntity} from "../interfaces/IModuleManager.sol";
+
+// Hook types:
+// Exec hook: bools for hasPre, hasPost
+// Validation hook: no bools
+
+// Hook fields:
+// module address
+// entity ID
+// hook type
+// if exec hook: hasPre, hasPost
+
+// Hook config is a packed representation of a hook function and flags for its configuration.
+// Layout:
+// 0xAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA________________________ // Address
+// 0x________________________________________BBBBBBBB________________ // Entity ID
+// 0x________________________________________________CC______________ // Type
+// 0x__________________________________________________DD____________ // exec hook flags
+//
+
+// Hook types:
+// 0x00 // Exec (selector and validation associated)
+// 0x01 // Validation
+
+// Exec hook flags layout:
+// 0b000000__ // unused
+// 0b______A_ // hasPre
+// 0b_______B // hasPost
+
+library HookConfigLib {
+    // Hook type constants
+    // Exec has no bits set
+    bytes32 internal constant _HOOK_TYPE_EXEC = bytes32(uint256(0));
+    // Validation has 1 in the 25th byte
+    bytes32 internal constant _HOOK_TYPE_VALIDATION = bytes32(uint256(1) << 56);
+
+    // Exec hook flags constants
+    // Pre hook has 1 in 2's bit in the 26th byte
+    bytes32 internal constant _EXEC_HOOK_HAS_PRE = bytes32(uint256(1) << 49);
+    // Post hook has 1 in 1's bit in the 26th byte
+    bytes32 internal constant _EXEC_HOOK_HAS_POST = bytes32(uint256(1) << 48);
+
+    function packValidationHook(ModuleEntity _hookFunction) internal pure returns (HookConfig) {
```

Comment:
Is there a reason the function parameters in this library are prefixed with an underscore?

Reply:
Yeah, the individual accessor methods like `module()`, `entityId()`, etc, are also visible within the other internal functions, so if the parameters would be named with the underscore (i.e. just `module`), then they would shadow the inner functions. It's not necessarily a bad thing since we don't really need to access those functions, but I just wanted to avoid any variable shadowing to prevent the compiler warning.